### PR TITLE
Databases/week2

### PR DESCRIPTION
**Pull request by AnilMula**

### CODE REVIEW

File: databases/week2/MySQl-Homework-Week2 part4.sql [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+-- -----------------------------------------------------
+-- Table `farm`.`farmers`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `farm`.`farmers` (
+  `id` VARCHAR(45) NOT NULL,
+  `farmer_id` INT NOT NULL,
+  `farmer_name` VARCHAR(45) NOT NULL,
+  `farmer_address` VARCHAR(45) NOT NULL,
+  `farmer_mobile` INT(10) NOT NULL,
+  `animal_id` INT NULL,
```
Comment (babakbandpey) [inline databases/week2/MySQl-Homework-Week2 part4.sql:26]: remove from this table and put it in an intermediary table which connects the farmer to many animal. This one allows you to assign only one animal to each farmer.

File: databases/week2/MySQl-Homework-Week2 part2.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use school;
+
+-- create index on name column of the student table
+create index name_index on student(name);
+
+-- add a new column to the class table nmed status with enum(not-started, ongoing, finished)
+use school;
+alter table class add status enum('not-started', 'ongoing', 'finished') not null;
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply databases/week2/MySQl-Homework-Week2 part2.sql]: added default value 


alter table class add status enum('not-started', 'ongoing', 'finished') not null default 'not-started';

File: databases/week2/MySQl-Homework-Week2 part3.sql [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+inner join user on task.user_id = user.id 
+inner join status on task.status_id = status.id
+where user.name = 'Donald Duck' and status.id = 1;
+
+-- Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+select title, created 
+from task
+inner join user on task.user_id = user.id
+where user.name = 'Maryrose Meadows' and task.created like '%-09-%';
+
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply databases/week2/MySQl-Homework-Week2 part3.sql:21]: used month(created)

File: databases/week2/MySQl-Homework-Week2 part4.sql [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  UNIQUE INDEX `animal_id_UNIQUE` (`animal_id` ASC) VISIBLE,
+  UNIQUE INDEX `animal_name_UNIQUE` (`animal_name` ASC) VISIBLE,
+  INDEX `fk_animals_farmers_idx` (`farmers_id` ASC) VISIBLE,
+  CONSTRAINT `fk_animals_farmers`
+    FOREIGN KEY (`farmers_id`)
+    REFERENCES `farm`.`farmers` (`id`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION)
+ENGINE = InnoDB;
+
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply databases/week2/MySQl-Homework-Week2 part4.sql:41]: schema changed

File: databases/week2/MySQl-Homework-Week2 part4.sql [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  `animals_animal_id` INT NOT NULL,
+  PRIMARY KEY (`id`),
+  INDEX `fk_barns_animals1_idx` (`animals_animal_id` ASC) VISIBLE,
+  CONSTRAINT `fk_barns_animals1`
+    FOREIGN KEY (`animals_animal_id`)
+    REFERENCES `farm`.`animals` (`animal_id`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION)
+ENGINE = InnoDB;
+
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply databases/week2/MySQl-Homework-Week2 part4.sql:76]: schema changed

File: databases/week2/MySQl-Homework-Week2 part4.sql [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+-- -----------------------------------------------------
+CREATE SCHEMA IF NOT EXISTS `farm` DEFAULT CHARACTER SET utf8 ;
+USE `farm` ;
+
+-- -----------------------------------------------------
+-- Table `farm`.`farmers`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `farm`.`farmers` (
+  `id` VARCHAR(45) NOT NULL,
+  `farmer_id` INT NOT NULL,
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply databases/week2/MySQl-Homework-Week2 part4.sql]: schema changed

File: databases/week2/MySQl-Homework-Week2 part4.sql [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  `farmer_name` VARCHAR(45) NOT NULL,
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply databases/week2/MySQl-Homework-Week2 part4.sql]: schema changed

File: databases/week2/MySQl-Homework-Week2 part4.sql [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  `farmer_address` VARCHAR(45) NOT NULL,
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply databases/week2/MySQl-Homework-Week2 part4.sql]: removed

File: databases/week2/MySQl-Homework-Week2 part4.sql [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  `farmer_mobile` INT(10) NOT NULL,
```
[the remark this answers is not in the document]
Comment (AnilMula) [reply databases/week2/MySQl-Homework-Week2 part4.sql]: removed

File: databases/week2/MySQl-Homework-Week2 part4.sql (hunk 1 above)
Comment (AnilMula) [reply databases/week2/MySQl-Homework-Week2 part4.sql]: schema changed

**Pull request merged**